File: 0650/solution.ts
function minSteps(n: number): number {
  let factor = 2;
  let res = 0;
  let root = Math.ceil(Math.sqrt(n));
  while (factor <= root) {
    while (n % factor == 0) {
      res += factor;
      n /= factor;
    }
    factor += 1;
  }
  return n == 1 ? res : res + n;
}

File: 0650/solution.go
package main

import (
	"math"
)

func minSteps(n int) int {
    res := 0
	root := int(math.Ceil(math.Sqrt(float64(n))))
	factor := 2
	
	for factor <= root {
		for n%factor == 0 {
			res += factor
			n /= factor
		}
		factor++
	}
	
	if n > 1 {
		res += n
	}
	
	return res
}